bundles/io/org.eclipse.smarthome.io.net.security/src/main/java/org/eclipse/smarthome/io/net/security/api/NetworkServerTls.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.net.security.api;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.security.GeneralSecurityException;
+
+import javax.net.ssl.SSLContext;
+
+import org.eclipse.californium.elements.Connector;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * An implementation of this interface would return SSLContext and DTLSConnectors,
+ * pre-configured for secure transmissions.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public interface NetworkServerTls {
+    /**
+     * Creates a new SSLContext with the given context or default context if the given one can't be found.
+     * This method is blocking and can take a while to resolve. If the default context is not existing yet,
+     * a provider is required to create a self-signed certificate and public/private-key pair.
+     *
+     * If not configured different, the default algorithm will be RSA. The signing algorithm will be SHA1.
```

Comment:
See other comment about SHA1. The RSA bits are configurable, why should they be mentioned in the javadoc?

Reply:
You are stating what the default is and the default for a crypto algorithm includes its bit size, that's why I think you should mention what you configured as a default.

Reply:
I mention the defaults further down in the configuration section, I will remove the paragraph therefore (I'm afraid it will get inconsistent if we change the algorithms and bits sizes at some point),